### PR TITLE
ENH add initialize method for input loader

### LINKED ISSUE #1162: Add an `initialize` method for InputLoader
@beckermr found a need for an input object to run some code directly after being made, but which can't be part of the regular `init_func`, and `setupImage` is already too late.  The solution we came up with is to add an `initialize` method that we would call directly after building the input item.

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds the `initialize` method for inputs. 

todo:

 - [x] make sure it solves my problem
 - [x] write a test

closes #1162

### CODE REVIEW

**Comment by beckermr** [comment]
See https://github.com/LSSTDESC/seacliff/pull/4 for an example use case.

**Comment by beckermr** [comment]
@rmjarvis This one is ready for review!

**Review by rmjarvis — CHANGES_REQUESTED** [review]
Looks good.  Just one request to add a test that checks this when there are two input objs being processed.

**Comment by rmjarvis** [comment]
Code looks fine.  Just one thing I didn't notice the first time.  Could you please rebase this onto main, rather than releases/2.3 and redirect the target of the merge?  This is a feature addition, so it shouldn't really go on the 2.3 series.

**Comment by beckermr** [comment]
Will rebase but need to brush up on my git fu unless you have a quick oneliner for me. :)

**Comment by rmjarvis** [comment]
(I always forget how to do the upsteam thing for forked repos.  It's easier if you started from a branch in GalSim-developers.)

**Comment by beckermr** [comment]
This worked for me: https://medium.com/@gabriellamedas/git-rebase-and-git-rebase-onto-a6a3f83f9cce

TL;DR

```
git rebase --onto main releases/2.3 init-input
```

Moved all commits on `init-input` ahead of `releases/2.3` to `main`.

**Comment by rmjarvis** [comment]
Oh, right.  That rebases onto your forked main, which is fine if that's current with the upstream main.

**Comment by beckermr** [comment]
Awesome. FYI I cannot merge this PR so you'll have to do it once the tests pass.

File: tests/test_config_input.py
Comment (rmjarvis) [inline tests/test_config_input.py:1]:
```suggestion
# Copyright (c) 2012-2022 by the GalSim developers team on GitHub
```

**Comment by beckermr** [comment]
This is ready to merge!

**Pull request merged**
